packages/suite-desktop/src-electron/modules/request-interceptor.ts

```diff
@@ -0,0 +1,27 @@
+/**
+ * Request Interceptor
+ * This module intercepts requests from electron nodejs main process and
+ * lets request-manager interceptor knows if Tor is enable so it has to use Tor or not.
+ *
+ *
+ * Differences from request-filter module is that it intercepts all requests from electron nodejs main process,
+ * whereas request-filter logs and filters allowed requests from electron renderer process.
+ */
+import { createInterceptor, InterceptedEvent } from '@trezor/request-manager';
+import { Module } from './index';
+
+const init: Module = ({ store }) => {
+    const { logger } = global;
+
+    const handleRequest = (event: InterceptedEvent) => {
+        if (event.method && event.details) {
+            logger.debug('request-interceptor', `${event.method} - ${event.details}`);
+        }
+        // We need to let the interceptor know that user wants to use Tor or not.
+        return store.getTorSettings().running;
+    };
+
+    createInterceptor(handleRequest);
+};
```

Comment:
I understand that with each invocation of handleRequest you are reading rom store to get settings of tor, right? It seems reasonable. 

What I don't like about it is the level of indirection here. Some module in electron is returing boolean which is handled on some totaly different place. So far I am only brainstorming and the first thing that came across my mind was something like this

```
const options = {
        handler: (event: InterceptedEvent) => {
            if (event.method && event.details) {
                logger.debug('request-interceptor', `${event.method} - ${event.details}`);
            }
        },
        settings: {
            torRunning: store.getTorSettings().running   
        }
    }

    createInterceptor(options);
```

Reply:
> settings: {
>             torRunning: store.getTorSettings().running   
>         }

The issue with this approach is that setting would be only read when the app starts but Tor status can be disable dynamically by the user after starting the app.

But maybe it could be like:
```javascript
const options = {
        handler: (event: InterceptedEvent) => {
            if (event.method && event.details) {
                logger.debug('request-interceptor', `${event.method} - ${event.details}`);
            }
        },
        getIsTorRunning: () => store.getTorSettings().running ;
 }

createInterceptor(options);
```

Or similar.

Reply:
Understood. It still looks better I think. More readable. I would do it like this. :pray: